react-markdown/react-markdown.d.ts

```diff
@@ -0,0 +1,182 @@
+// Type definitions for react-markdown
+// Project: https://github.com/rexxars/react-markdown
+// Definitions by: Greg Rosenbaum <https://github.com/GregRos>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+///<reference types="react"/>
+declare namespace ReactMarkdown {
```

Comment:
In the `types-2.0` branch we prefer an external module style. Here's how it works:

Instead of using the `React` namespace, use:

```ts
import { Component, ReactChild, ReactElement } from "react";
```

Instead of declaring a `ReactMarkdown` namespace, use:

```ts
declare namespace MarkdownComponent {
    ...
}
```

And put all of your types *except* `MarkdownComponent` in there.

Then on the top level, do:

```ts
declare class MarkdownComponent extends Component<MarkdownComponent.ReactMarkdownProps, {}> { }
export = MarkdownComponent;
```

The namespace `MarkdownComponent` will be merged with the class.
This way you can use your declared types like so:

```ts
import ReactMarkdown = require("react-markdown");
const x: ReactMarkdown.NodeName = "HtmlBlock";
```